### ISSUE TITLE
Could not cast value of type 'StreamChat.CurrentUserDTO' [...] to 'StreamChat.CurrentUserDTO'

### ISSUE THREAD

**Issue by ratkins** [GetStream/stream-chat-swift#1572]
We're getting this one reliably when running our acceptance tests:

```
Thread 11 Queue : NSManagedObjectContext 0x600003f25a00 (serial)
#0	0x00000001c7f8de50 in __pthread_kill ()
#1	0x00000001c7fdd8d4 in pthread_kill ()
#2	0x000000010a7203d8 in abort ()
#3	0x000000010a0b4b4c in swift::fatalError(unsigned int, char const*, ...) ()
#4	0x000000010a0ad21c in swift::swift_dynamicCastFailure(void const*, char const*, void const*, char const*, char const*) ()
#5	0x000000010a0ad294 in swift::swift_dynamicCastFailure(swift::TargetMetadata<swift::InProcess> const*, swift::TargetMetadata<swift::InProcess> const*, char const*) ()
#6	0x000000010a0adf98 in swift_dynamicCastClassUnconditional ()
#7	0x00000001075fd2e8 in static CurrentUserDTO.loadOrCreate(context:) at /Users/ratkins/Library/Developer/Xcode/DerivedData/ios-app-asfoctiwvzwspdgwziefzxgdlvzr/SourcePackages/checkouts/stream-chat-swift/Sources/StreamChat/Database/DTOs/CurrentUserDTO.swift:59
#8	0x00000001075fd358 in NSManagedObjectContext.saveCurrentUser(payload:) at /Users/ratkins/Library/Developer/Xcode/DerivedData/ios-app-asfoctiwvzwspdgwziefzxgdlvzr/SourcePackages/checkouts/stream-chat-swift/Sources/StreamChat/Database/DTOs/CurrentUserDTO.swift:66
#9	0x00000001075ff928 in protocol witness for CurrentUserDatabaseSession.saveCurrentUser(payload:) in conformance NSManagedObjectContext ()
#10	0x000000010762a720 in DatabaseSession.saveEvent(payload:) at /Users/ratkins/Library/Developer/Xcode/DerivedData/ios-app-asfoctiwvzwspdgwziefzxgdlvzr/SourcePackages/checkouts/stream-chat-swift/Sources/StreamChat/Database/DatabaseSession.swift:310
#11	0x00000001076e5138 in EventDataProcessorMiddleware.handle(event:session:) at /Users/ratkins/Library/Developer/Xcode/DerivedData/ios-app-asfoctiwvzwspdgwziefzxgdlvzr/SourcePackages/checkouts/stream-chat-swift/Sources/StreamChat/WebSocketClient/EventMiddlewares/EventDataProcessorMiddleware.swift:15
#12	0x00000001076e6388 in protocol witness for EventMiddleware.handle(event:session:) in conformance EventDataProcessorMiddleware ()
#13	0x00000001076e64b8 in Array<τ_0_0>.process(event:session:) at /Users/ratkins/Library/Developer/Xcode/DerivedData/ios-app-asfoctiwvzwspdgwziefzxgdlvzr/SourcePackages/checkouts/stream-chat-swift/Sources/StreamChat/WebSocketClient/EventMiddlewares/EventMiddleware.swift:31
#14	0x0000000107902504 in closure #1 in EventNotificationCenter.process(_:completion:) at /Users/ratkins/Library/Developer/Xcode/DerivedData/ios-app-asfoctiwvzwspdgwziefzxgdlvzr/SourcePackages/checkouts/stream-chat-swift/Sources/StreamChat/Workers/EventNotificationCenter.swift:59
#15	0x0000000107625038 in closure #1 in DatabaseContainer.write(_:completion:) at /Users/ratkins/Library/Developer/Xcode/DerivedData/ios-app-asfoctiwvzwspdgwziefzxgdlvzr/SourcePackages/checkouts/stream-chat-swift/Sources/StreamChat/Database/DatabaseContainer.swift:177
#16	0x00000001074949fc in thunk for @escaping @callee_guaranteed () -> () ()
#17	0x000000011fc42e8c in developerSubmittedBlockToNSManagedObjectContextPerform ()
#18	0x000000010917b6c8 in _dispatch_client_callout ()
#19	0x0000000109182800 in _dispatch_lane_serial_drain ()
#20	0x0000000109183384 in _dispatch_lane_invoke ()
#21	0x000000010918db44 in _dispatch_workloop_worker_thread ()
#22	0x00000001c7fda028 in _pthread_wqthread ()
```

This in the console:
```
Could not cast value of type 'StreamChat.CurrentUserDTO' (0x600002026e90) to 'StreamChat.CurrentUserDTO' (0x1079b3998).
dyld4 config: DYLD_ROOT_PATH=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot DYLD_LIBRARY_PATH=/Users/ratkins/Library/Developer/Xcode/DerivedData/ios-app-asfoctiwvzwspdgwziefzxgdlvzr/Build/Products/Debug-iphonesimulator DYLD_INSERT_LIBRARIES=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/usr/lib/libMainThreadChecker.dylib DYLD_FRAMEWORK_PATH=/Users/ratkins/Library/Developer/Xcode/DerivedData/ios-app-asfoctiwvzwspdgwziefzxgdlvzr/Build/Products/Debug-iphonesimulator:/Users/ratkins/Library/Developer/Xcode/DerivedData/ios-app-asfoctiwvzwspdgwziefzxgdlvzr/Build/Products/Debug-iphonesimulator/PackageFrameworks
Could not cast value of type 'StreamChat.CurrentUserDTO' (0x600002026e90) to 'StreamChat.CurrentUserDTO' (0x1079b3998).
CoreSimulator 776.4 - Device: iPhone 13 (AC2688B9-3504-4F82-A410-7639FA0D11C3) - Runtime: iOS 15.0 (19A339) - DeviceType: iPhone 13
```

StreamChat 4.1.0 via SPM
Xcode Version 13.1 (13A1030d)
Simulator Version 13.1 (970) running iOS 15.0

**Comment by b-onc** [comment]
Hello @ratkins 

We're having difficulties reproducing your setup. Would you mind sharing a simple sample where this issue can be reproduced?

**Comment by ratkins** [comment]
I'm not sure I have time to build a reduced test case, but maybe we could pair and I could show you our setup directly? Send me an email: my GitHub username @beams.fm.

**Comment by b-onc** [comment]
Hello,

We've released 4.6.0 which should resolve this issue.

I'm closing this ticket since we moved the discussion to our internal support. Ticket number #16649
